### PR TITLE
feat(logger_level_reconfigure_plugin): use node interface and some cosmetic updates

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Update logger level reconfigure plugin.

- Use node interface following https://github.com/autowarefoundation/autoware.universe/pull/5146
- Move hard-corded configuration into yaml file
- Make the panel scrollable
- Change the button color based on the logger level

Special thanks to Chat-GPT. This revision is mostly done by his implementation.

https://github.com/autowarefoundation/autoware.universe/assets/21360593/7613845c-7d72-4222-accb-650b679aedc1

## Related links

None

## Tests performed

run psim

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp [hunk 1]
```diff
@@ -26,102 +30,98 @@ LoggingLevelConfigureRvizPlugin::LoggingLevelConfigureRvizPlugin(QWidget * paren
...
+}
+
+// Calculate the maximum width among all target_module_name.
+int LoggingLevelConfigureRvizPlugin::getMaxModuleNameWidth(QLabel * label)
+{
+  int max_width = 0;
+  QFontMetrics metrics(label->font());
+  for (const auto & item : node_logger_map_) {
+    const auto & target_module_name = item.first;
+    int width = metrics.horizontalAdvance(target_module_name);
```
Comment (kosuke55) [inline common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp:104]:
```suggestion
    const int width = metrics.horizontalAdvance(target_module_name);
```

**Review by kosuke55 — APPROVED** [review]

File: common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp (hunk 1 above)
Comment (kosuke55) [reply common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp:104]: or std::max

File: common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp [hunk 2]
```diff
@@ -26,102 +30,98 @@ LoggingLevelConfigureRvizPlugin::LoggingLevelConfigureRvizPlugin(QWidget * paren
...
+    if (width > max_width) {
+      max_width = width;
+    }
```
Comment (kosuke55) [inline common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp:107]: or
```suggestion
    max_width = std::max(metrics.horizontalAdvance(target_module_name), max_width);
```

File: common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp [hunk 3]
```diff
@@ -26,102 +30,98 @@ LoggingLevelConfigureRvizPlugin::LoggingLevelConfigureRvizPlugin(QWidget * paren
...
-    std::cerr << "load logger in " << container_name.toStdString() << ": result = " << result
-              << std::endl;
+  int max_width = 0;
+  QFontMetrics metrics(label->font());
+  for (const auto & item : node_logger_map_) {
+    const auto & target_module_name = item.first;
+    const int width = metrics.horizontalAdvance(target_module_name);
+    if (width > max_width) {
+      max_width = width;
+    }
```
Comment (TakaHoribe) [inline common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp:107]:
```suggestion
    max_width = std::max(metrics.horizontalAdvance(target_module_name), max_width);
```

File: common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp (hunk 2 above)
Comment (TakaHoribe) [reply common/tier4_logging_level_configure_rviz_plugin/src/logging_level_configure.cpp]: Thank you!

**Pull request merged**